digital_land/update.py

```diff
@@ -1,44 +1,42 @@
 from datetime import datetime, date
-from digital_land.register import Item
-from digital_land.collection import LogRegister, EndpointRegister, SourceRegister
-from pathlib import Path
-import hashlib
+
+from .register import Item, hash_value
+from .collection import Collection
+from .schema import Schema
 
 
 def get_source_endpoint_fieldnames():
-    return set(EndpointRegister.fieldnames + SourceRegister.fieldnames)
+    return set(Schema("endpoint").fieldnames + Schema("source").fieldnames)
 
 
 def get_failing_endpoints_from_registers(
-    log_path, endpoints_dir, first_date, last_date=date.today()
+    log_path, collection_directory, first_date, last_date=date.today()
```

Comment:
a bit of me wonders if we should remove the Collection dependency from this file too, and pass in a fully built collection object from cli.py (same in `add_new_source_endpoint`)